### PR TITLE
Fixed #34026 -- Fixed WKBReader.read() crash on string input.

### PR DESCRIPTION
**Pull request by select-case**
Changes made according to #34026 WKBReader.read() crashes on strings.

### CODE REVIEW

File: django/contrib/gis/geos/prototypes/io.py [hunk 1]
```diff
@@ -169,8 +169,11 @@ def read(self, wkb):
         if isinstance(wkb, memoryview):
             wkb_s = bytes(wkb)
             return wkb_reader_read(self.ptr, wkb_s, len(wkb_s))
-        elif isinstance(wkb, (bytes, str)):
+        elif isinstance(wkb, bytes):
             return wkb_reader_read_hex(self.ptr, wkb, len(wkb))
+        elif isinstance(wkb, str):
+            wkb_s = wkb.encode()
+            return wkb_reader_read_hex(self.ptr, wkb_s, len(wkb))
```
Comment (benoit9126) [inline django/contrib/gis/geos/prototypes/io.py:176]: I would have written `len(wkb_s)` as last argument of `wkb_reader_read_hex`

Comment (select-case) [reply django/contrib/gis/geos/prototypes/io.py]: I have made that change.

File: tests/gis_tests/geos_tests/test_io.py [hunk 2]
```diff
@@ -67,6 +67,15 @@ def test03_wkbreader(self):
         for geom in (g1, g2):
             self.assertEqual(ref, geom)
 
+        str1 = "000000000140140000000000004037000000000000"
+        wkb_1 = memoryview(binascii.a2b_hex(str1))
+        ref_1 = GEOSGeometry(str1)
```
Comment (claudep) [inline tests/gis_tests/geos_tests/test_io.py:72]: I don't think we need to recreate `wkb_1` and `ref_1` as they will be identical to `wkb` and `ref`. So I would simply add `str1` above and test it in the same loop above (with only `g3` added). We might rename the variables `hex_bin` and hex_str`.

**Review by claudep — CHANGES_REQUESTED** [review]

File: tests/gis_tests/geos_tests/test_io.py (hunk 2 above)
Comment (select-case) [reply tests/gis_tests/geos_tests/test_io.py]: Made the asked for change

**Review by claudep — APPROVED** [review]
Thanks, exactly what I meant!

**Comment by felixxm** [comment]
@select-case Thanks :+1: Welcome aboard :boat:

**Pull request merged**